torch/csrc/lazy/core/tensor_impl.h

```diff
@@ -57,9 +57,10 @@ class TORCH_API LTCTensorImpl final : public c10::TensorImpl {
 
  private:
   void setup_size_properties();
+  c10::SymIntArrayRef setup_sym_sizes() const;
 
   LazyTensorPtr tensor_;
-  std::vector<c10::SymInt> sym_sizes_;
+  mutable c10::optional<std::vector<c10::SymInt>> sym_sizes_;
```

Comment:
hmm, mutable since we expect to change this on const tensors? i guess thats ok? but i'm not sure if it would be seen as bad practice

Reply:
Oh I think this is one of the acceptable usage of `sym_sizes_`. We aren't changing the logical state of an object when we query sizes, we are just caching the result. It's usually suggested to use `mutable` rather than casting to a non_const reference. https://users.rust-lang.org/t/use-cases-for-interior-mutability/21341/2 lmao sorry for the rust reference :)